### PR TITLE
build: switch to yarn

### PR DESCRIPTION
**Pull request by chrismwendt**
See https://github.com/sourcegraph/sourcegraph/issues/13181

semantic-release with yarn info https://github.com/semantic-release/semantic-release/blob/468f8a8bda01ea1d8202f9f405ad77f8adbf2a9a/docs/support/FAQ.md#can-i-use-semantic-release-with-yarn

### CODE REVIEW

**Comment by ijsnow** [comment]
I'm fine with this but don't know of the required changes around CI and auto deployments cc @felixfbecker

File: src/libs/github/inject.tsx [hunk 1]
```diff
@@ -132,6 +132,7 @@ function injectCodeIntelligence(): void {
 
     if (isSingleCodeFile && useExtensions && filePath) {
         extensionsContextController = createExtensionsContextController(sourcegraphUrl)
+        console.log('calling createController')
```
[the remark this answers is not in the document]
Comment (chrismwendt) [reply src/libs/github/inject.tsx]: Oops, I removed the stray console.log.

**Comment by chrismwendt** [comment]
There was an uglifyjs problem, and pinning the versions of uglify-js and uglifyjs-webpack-plugin to the old versions caused some other error, so I left the versions upgraded and fixed a resulting type error by replacing our `src/types/uglifyjs-webpack-plugin/index.d.ts` with `@types/uglifyjs-webpack-plugin`.

<details>

Error with old version of uglifyjs:

```
ERROR in options.bundle.js from UglifyJs
Unexpected token: operator (>) [options.bundle.js:2368,25]
```

</details>

**Comment by sqs** [comment]
I just removed uglify in my branch because I saw the same problem. It is not needed for an oss browser extension.

**Comment by felixfbecker** [comment]
Did you use the uglify/webpack config from the webapp? It works fine there and should be up to date.

**Comment by felixfbecker** [comment]
@sqs uglify does more than just "uglification", it is also responsible for removing dead code. For example, React will use the development build if `if (process.env.NODE_ENV === 'development')`, which webpack will replace with `if (false)`, which Uglify will then remove.

**Comment by chrismwendt** [comment]
@ijsnow Cypress passes locally but fails in CI. Can I somehow view the screenshots from https://travis-ci.org/sourcegraph/browser-extensions/jobs/428345800 ? Or is there another way to troubleshoot this?

**Comment by sqs** [comment]
@chrismwendt: I am seeing the same thing on my PR. @ijsnow will look into it shortly. No need to try to troubleshoot yourself (I went a bit but was unsuccessful).

**Comment by ijsnow** [comment]
https://github.com/sourcegraph/browser-extensions/pull/161/commits/26029be0322e07f7538d18a3e88a0982d26c0be5 

That commit doesn't fix the CI problem but it uses the latest stable version of yarn (1.9.4) rather than what travis comes out of the box with (1.3.2)

**Comment by chrismwendt** [comment]
@sqs I will wait until you merge in https://github.com/sourcegraph/browser-extensions/pull/165 and https://github.com/sourcegraph/browser-extensions/pull/162 before merging this in.

**Comment by sqs** [comment]
@chrismwendt I just merged #165. You should merge this before I merge #162 and don't worry, I will rebase #162 on the new master.

**Comment by sqs** [comment]
(so you should merge this branch now)

**Comment by chrismwendt** [comment]
Ok, I rebased and now I'm fixing the build locally.

**Comment by chrismwendt** [comment]
Pushed, waiting on CI now.

**Pull request merged**